capellambse_context_diagrams/serializers.py

```diff
@@ -71,17 +71,11 @@ def make_diagram(self, data: _elkjs.ELKOutputData) -> aird.Diagram:
         for child in data["children"]:
             self.deserialize_child(child, aird.Vector2D(), None)
 
-        space_checkers = {
-            "box": self.check_boxlabel_space,
-            "edge": self.check_edgelabel_space,
-            "symbol": lambda _: None,  # No need since label is outside
-        }
         for element in self._cache.values():
-            if not element.port:
-                space_checkers[element.JSON_TYPE](element)
-            if element.JSON_TYPE != "edge":
-                self.check_viewBox_space(element)
+            if element.JSON_TYPE == "box":
+                self.check_boxlabel_space(element)
```

Comment:
Why is this still necessary? I thought the point of using `calculate_viewport` was that we don't need to do any of this here anymore.

Reply:
Glad that you ask. This is especially needed for ContextDiagrams of interfaces since Functions are placed inside a Component rectangle instead of just the canvas. If the Component has a large label (multilines) then the first Function rectangle would overlap its parent label.

We stretch the size to the north in that case... which can lead to edges overlapping this stretched area, since serialization is the last step.

Is there functionality on Boxes to reach the same behaviour?

Reply:
A nice way of using ELK's hidden box sizing feature by children/label box sizes (which makes parent box sizes useless) was found. This will let us remove the last label space checker and even fix a bug of occuring diagram element overlaps of adjusted boxes.